### PR TITLE
DOC-837 Allow trial licenses in RPCN

### PR DESCRIPTION
**Pull request by asimms41**
## Description

Resolves [DOC-837](https://redpandadata.atlassian.net/browse/DOC-837)
Review deadline: 9th December

Related PR: [https://github.com/redpanda-data/docs/pull/900](https://github.com/redpanda-data/docs/pull/900)

## Page previews

https://deploy-preview-120--redpanda-connect.netlify.app/redpanda-connect/get-started/licensing/

## Checks

- [ ] New feature
- [x] Content gap
- [ ] Support Follow-up
- [ ] Small fix (typos, links, copyedits, etc)

[DOC-837]: https://redpandadata.atlassian.net/browse/DOC-837?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by Jeffail — APPROVED** [review]

File: modules/get-started/pages/licensing.adoc [hunk 1]
```diff
@@ -1,17 +1,13 @@
 = Enterprise Licensing 
 :description: Learn how to purchase and apply an Enterprise Edition license.
 
-Anyone can use the Certified and Community connectors, but you need an Enterprise Edition license to use Enterprise connectors.
+Anyone can use the Certified and Community connectors, but you need a valid Enterprise Edition license to use Enterprise connectors. 
 
-== Purchase an Enterprise Edition license
-
-You can purchase and download a license from the https://www.redpanda.com/upgrade[Redpanda website^].
+You can either https://www.redpanda.com/upgrade[upgrade to an Enterprise Edition license^], or http://redpanda.com/try-enterprise[generate a trial license key^] that's valid for 30 days.
```
Comment (JakeSCahill) [inline modules/get-started/pages/licensing.adoc:6]: Maybe put this under a 'Get an enterprise license' heading? It's a user goal that may be searched for.

File: modules/get-started/pages/licensing.adoc [hunk 2]
```diff
@@ -1,17 +1,13 @@
...
 
 == Apply a license key to Redpanda Connect
 
-Redpanda Connect checks the license key status at runtime. If the license key is unavailable or has expired, you are blocked from using Enterprise connectors. You can view the license key’s expiration date at any time in the Redpanda Connect logs.
-
-Once you have downloaded a license key, you can apply it in the following ways: 
+Once you have downloaded the license key, you can apply it in the following ways:
```
Comment (JakeSCahill) [inline modules/get-started/pages/licensing.adoc:10]:
```suggestion
When you have downloaded the license key, you can apply it in the following ways: 
```

File: modules/get-started/pages/whats-new.adoc [hunk 3]
```diff
@@ -6,11 +6,15 @@ This topic includes new content added from version 4.29.0 onwards.
 
 For a full list of product updates, and details of updates prior to version 4.29.0, see the https://github.com/redpanda-data/connect/releases[Redpanda Connect release notes^]. For new Cloud-specific features, see xref:redpanda-cloud:get-started:whats-new-cloud.adoc[].
 
+
+== Version 4.44.0
+
+This release reintroduces trial licenses for Self-Managed Redpanda Connect.
```
Comment (JakeSCahill) [inline modules/get-started/pages/whats-new.adoc:12]:
```suggestion
This release introduces support for trial licenses in Redpanda Connect Self-Managed.
```

File: modules/get-started/pages/licensing.adoc [hunk 4]
```diff
@@ -33,3 +28,9 @@ Replace the following placeholders:
 
 - `<license-string>`: The full license key string.
 - `<connect-configuration>`: The name of your Redpanda Connect configuration file.
+
+== License Verification
```
Comment (JakeSCahill) [inline modules/get-started/pages/licensing.adoc:32]:
```suggestion
== Verify a license
```

File: modules/get-started/pages/licensing.adoc (hunk 2 above)
Comment (JakeSCahill) [reply modules/get-started/pages/licensing.adoc:10]: https://github.com/redpanda-data/docs-site/blob/main/meta-docs/STYLE-GUIDE.adoc#do-not-use-word-list

File: modules/get-started/pages/whats-new.adoc (hunk 3 above)
Comment (JakeSCahill) [reply modules/get-started/pages/whats-new.adoc:12]: I don't think RPCN ever introduced support for trial licenses.

File: modules/get-started/pages/licensing.adoc (hunk 1 above)
Comment (asimms41) [reply modules/get-started/pages/licensing.adoc:6]: Updated

File: modules/get-started/pages/licensing.adoc (hunk 2 above)
Comment (asimms41) [reply modules/get-started/pages/licensing.adoc:10]: Updated

File: modules/get-started/pages/whats-new.adoc (hunk 3 above)
Comment (asimms41) [reply modules/get-started/pages/whats-new.adoc:12]: Updated

File: modules/get-started/pages/licensing.adoc (hunk 4 above)
Comment (asimms41) [reply modules/get-started/pages/licensing.adoc:32]: Updated

**Review by JakeSCahill — APPROVED** [review]

**Pull request merged**